design/ceph/osd-scrubbing-schedule.md

```diff
@@ -0,0 +1,45 @@
+# OSD Scrubbing Schedule
+
+## Goal
+
+Adding a simple way to configure the Ceph OSD Scrubbing schedule options via the CephCluster CRD.
+
+## Current Solution
+
+Users need to update the `rook-config-override` and add the [Ceph OSD Scrubbing config options](https://docs.ceph.com/en/latest/rados/configuration/osd-config-ref/#scrubbing) as they need it.
+
+## Proposed Solution
+
+Adding a new config structure to the CephCluster CRD under `.spec.storage` named `scrubbing:`.
+
+The config structure will contain the most important options for setting a custom OSD scrubbing schedule.
+
+```yaml
+spec:
+  # [...]
+  storage:
+    # [...]
+    scrubbing:
```

Comment:
Agreed, it has been a repeated topic that users would like the flexibility for arbitrary ceph settings, so they can avoid the ceph.conf overrides. If we make it generic, consider the following:
- The schema will be general to allow any setting, no specific schema for scrubbing or other settings
- The implementation could be with `ceph config assimilate-conf`, which would be efficient in case many settings are applied.

Reply:
Yeah I think it makes sense to add a system to set ceph config options on the cluster. I'm currently thinking though that we should keep the `rook-config-override` Configmap around for special cases where the config must already be set when  the first MON is just about to be created.

I should have an updated design doc ready by tomorrow.